src/EFCore.PG/Migrations/NpgsqlMigrationsSqlGenerator.cs

```diff
@@ -721,10 +706,37 @@ public virtual void Generate(NpgsqlDropDatabaseOperation operation, [CanBeNull]
             builder.AppendLine(';');
         }
 
+        #region Enum management
+
+        protected virtual void GenerateEnumStatements(
+                [NotNull] AlterDatabaseOperation operation,
+                [NotNull] IModel model,
+                [NotNull] MigrationCommandListBuilder builder)
+        {
+            foreach (var enumTypeToCreate in PostgresEnum.GetPostgresEnums(operation)
+                .Where(ne => PostgresEnum.GetPostgresEnums(operation.OldDatabase).All(oe => oe.Name != ne.Name)))
+            {
+                GenerateCreateEnum(enumTypeToCreate, model, builder);
+            }
+
+            foreach (var enumTypeToDrop in PostgresEnum.GetPostgresEnums(operation.OldDatabase)
+                .Where(oe => PostgresEnum.GetPostgresEnums(operation).All(ne => ne.Name != oe.Name)))
+            {
+                GenerateDropEnum(enumTypeToDrop, builder);
+            }
+
+            foreach (var enumTypeToAlter in from newEnum in PostgresEnum.GetPostgresEnums(operation)
+                join oldEnum in PostgresEnum.GetPostgresEnums(operation.OldDatabase) on newEnum.Name equals oldEnum.Name
+                select new { newEnum.Name, OldLabels = oldEnum.Labels, newLabels = newEnum.Labels })
+            {
+                // TODO: Some forms of enum alterations are actually supported... At least log...
```

Comment:
May be throw an exception?

Reply:
Done, also copied better LINQ construct from range.

Reply:
I don't know.... wouldn't that make standard code throw?